### PR TITLE
Don't unpack whl files by default in generic PythonPackage easyblock

### PR DESCRIPTION
**Pull request by Flamefire**
Especially with the introduction of https://github.com/easybuilders/easybuild-framework/pull/3618 Wheels may become more common as (THE) valid installation approach for Python packages via pip. Having to specify `unpack_sources = False` every time is annoying and feels like something is off.
This basically adds first-class support whl-files by NOT unpacking them by default.

This could be considered a breaking change for cases where whl files were used and they should be unpacked without explicitly specifying `unpack_sources = True` (as that was the default), but I don't think it was even possible to do that meaningfully, so I'd say the valid cases that this breaks are unicorn rare.

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _pkgconfig-1.5.1-GCCcore-8.2.0-python.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa9 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/10a023bf9b61d9427cb5e4a4fed609ec for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _pkgconfig-1.5.1-GCCcore-8.2.0-python.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/6e391ca923ed2e51057ea6df0fbbae63 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _pkgconfig-1.5.1-GCCcore-8.2.0-python.eb_ (partial log available at https://gist.github.com/9eb061334e41707a3b73c01bf65253f0)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/22265be0baffbbbb1c8d2664b5521988 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _pkgconfig-1.5.1-GCCcore-8.2.0-python.eb_ (partial log available at https://gist.github.com/cf88a695c759abe3149dd33dfa3a073c)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/859650686eab87d18c197ce207283f32 for a full test report.

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 1]
```diff
@@ -410,6 +411,17 @@ def prepare_python(self):
...
 
+    def _should_unpack_source(self):
+        """Determine whether we need to unpack the source(s)"""
+        unpack_sources = self.cfg.get('unpack_sources')
+        if unpack_sources is None:
+            src = self.src
+            if not isinstance(src, string_type):
+                src = src[0]['path']
+            _, ext = os.path.splitext(src)
+            unpack_sources = ext.lower() == '.whl'
```
Comment (boegel) [inline easybuild/easyblocks/generic/pythonpackage.py:422]: This should be `!=`, no? Unpack when we're *not* using a `.whl` as source?

Also, why not handle this on the framework side instead, similar to https://github.com/easybuilders/easybuild-framework/pull/3545 ?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/generic/pythonpackage.py (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyblocks/generic/pythonpackage.py:422]: Ouch, of course! How did this work then...?
I wouldn't mind, but am afraid, that this causes confusion and while we could unpack whl-files in general, changing the framework would disable that completely even if we wanted to unpack those in the future.

Comment (Flamefire) [reply easybuild/easyblocks/generic/pythonpackage.py]: Fixed the comparison

**Comment by Flamefire** [comment]
Rebased to resolve the conflict

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Ray-project-1.0.1-fosscuda-2019b-Python-3.7.4.eb_ 
 * **SUCCESS** _Python-3.8.6-GCCcore-10.2.0.eb_ 
 * **SUCCESS** _SciPy-bundle-2020.11-foss-2020b.eb_ 
 * **SUCCESS** _matplotlib-3.3.3-foss-2020b.eb_ 

Build succeeded for 4 out of 4 (4 easyconfigs in total)
node2660.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/89dea4a9bb5f0a1ab6d632f47ca8950f for a full test report.

**Review by boegel — APPROVED** [review]

**Pull request merged**